File: src/backend/AiObservability.Api/Models/Sources.cs
using System.Text.Json.Serialization;

namespace AiObservability.Api.Models;

public record SourceIntegration(
    string Id,
    string Name,
    SourceKind Kind,
    SourceHealth Health,
    string LastQueryAt,
    [property: JsonPropertyName("queries_24h")] int Queries24h,
    double ErrorRate,
    Dictionary<string, object> Config,
    bool Enabled = true,
    bool ComingSoon = false
);

File: src/frontend/src/types/sources.ts
export type SourceKind = 'app-insights' | 'grafana' | 'prometheus' | 'github';
export type SourceHealth = 'healthy' | 'degraded' | 'incident' | 'unknown';

export interface SourceIntegration {
  id: string;
  name: string;
  kind: SourceKind;
  health: SourceHealth;
  last_query_at: string;
  queries_24h: number;
  error_rate: number;
  config: Record<string, string | number | boolean>;
  enabled?: boolean;
  coming_soon?: boolean;
}
